OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitWebViewFragment.java

```diff
@@ -132,6 +152,64 @@ public void onPageLoadProgressChanged(WebView webView, int progress) {
         }
     }
 
+    private void evaluateJavascriptForiFrame() {
+        if (!TextUtils.isEmpty(unit.getBlockId()) && !evaluatediFrameJS) {
+            // execute js code to check an HTML block that contains an iframe
+            String javascript =
+                    "try {" +
+                            "    var top_div_list = document.querySelectorAll('div[data-usage-id=\"" + unit.getId() + "\"]');\n" +
+                            "    top_div_list.length == 1 && top_div_list[0].querySelectorAll(\"iframe\").length > 0;" +
+                            "} catch {" +
+                            "    false;" +
+                            "};";
+            authWebView.evaluateJavascript(javascript, value -> {
+                evaluatediFrameJS = true;
+                if (Boolean.parseBoolean(value)) {
+                    setupOpenInBrowserView();
+                } else {
+                    tvOpenBrowser.setVisibility(View.GONE);
+                }
+            });
+        }
+    }
+
+    private void setupOpenInBrowserView() {
+        tvOpenBrowser.setVisibility(View.VISIBLE);
+
+        String openInBrowserMessage = getString(R.string.open_in_browser_message) + " "
+                + getString(R.string.open_in_browser_text) + " " + AppConstants.ICON_PLACE_HOLDER;
+        SpannableString openInBrowserSpan = new SpannableString(openInBrowserMessage);
+
+        ImageSpan openInNewIcon = new ImageSpan(getContext(), R.drawable.ic_open_in_new);
+        openInBrowserSpan.setSpan(openInNewIcon, openInBrowserMessage.indexOf(AppConstants.ICON_PLACE_HOLDER),
+                openInBrowserMessage.length(), DynamicDrawableSpan.ALIGN_BASELINE);
+
+        ClickableSpan clickableSpan = new ClickableSpan() {
+            @Override
+            public void onClick(@NotNull View textView) {
+                BrowserUtil.open(getActivity(), unit.getWebUrl(), true);
+            }
+
+            @Override
+            public void updateDrawState(@NotNull TextPaint textPaint) {
+                textPaint.setUnderlineText(true);
+                super.updateDrawState(textPaint);
+            }
+        };
+        openInBrowserSpan.setSpan(clickableSpan,
+                openInBrowserMessage.indexOf(getString(R.string.open_in_browser_text)),
```

Comment:
Can we use a local parameter for this `openInBrowserMessage.indexOf(getString(R.string.open_in_browser_text))` as it is using on many places